### PR TITLE
Fix flakiness of testRemoteCleanupDeleteStale, bug fix in RemoteMetadataManifest and RemoteReadResult

### LINKED ISSUE #13968: [BUG] org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.testRemoteCleanupDeleteStale is flaky
### Describe the bug

The test case `org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.testRemoteCleanupDeleteStale` is flaky:

```
java.lang.AssertionError: Current number of manifest files: 30
	at __randomizedtesting.SeedInfo.seed([AF3CFB99BA9F2DEE:D90361535B7B574A]:0)
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.lambda$testRemoteCleanupDeleteStale$0(RemoteClusterStateCleanupManagerIT.java:127)
	at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1128)
	at org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.testRemoteCleanupDeleteStale(RemoteClusterStateCleanupManagerIT.java:121)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at org.opensearch.test.OpenSearchTestClusterRule$1.evaluate(OpenSearchTestClusterRule.java:369)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	Suppressed: java.lang.AssertionError: Current number of manifest files: 30
		at org.junit.Assert.fail(Assert.java:89)
		at org.junit.Assert.assertTrue(Assert.java:42)
		at org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.lambda$testRemoteCleanupDeleteStale$0(RemoteClusterStateCleanupManagerIT.java:127)
		at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1116)
		... 40 more
	Suppressed: java.lang.AssertionError: Current number of manifest files: 30
		at org.junit.Assert.fail(Assert.java:89)
		at org.junit.Assert.assertTrue(Assert.java:42)
		at org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.lambda$testRemoteCleanupDeleteStale$0(RemoteClusterStateCleanupManagerIT.java:127)
		at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1116)
		... 40 more
	Suppressed: java.lang.AssertionError: Current number of manifest files: 30
		at org.junit.Assert.fail(Assert.java:89)
		at org.junit.Assert.assertTrue(Assert.java:42)
		at org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.lambda$testRemoteCleanupDeleteStale$0(RemoteClusterStateCleanupManagerIT.java:127)
		at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1116)
		... 40 more
	Suppressed: java.lang.AssertionError: Current number of manifest files: 30
		at org.junit.Assert.fail(Assert.java:89)
		at org.junit.Assert.assertTrue(Assert.java:42)
		at org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.lambda$testRemoteCleanupDeleteStale$0(RemoteClusterStateCleanupManagerIT.java:127)
		at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1116)
		... 40 more
	Suppressed: java.lang.AssertionError: Current number of manifest files: 30
		at org.junit.Assert.fail(Assert.java:89)
		at org.junit.Assert.assertTrue(Assert.java:42)
		at org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.lambda$testRemoteCleanupDeleteStale$0(RemoteClusterStateCleanupManagerIT.java:127)
		at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1116)
		... 40 more
	Suppressed: java.lang.AssertionError: Current number of manifest files: 30
		at org.junit.Assert.fail(Assert.java:89)
		at org.junit.Assert.assertTrue(Assert.java:42)
		at org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.lambda$testRemoteCleanupDeleteStale$0(RemoteClusterStateCleanupManagerIT.java:127)
		at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1116)
		... 40 more
	Suppressed: java.lang.AssertionError: Current number of manifest files: 30
		at org.junit.Assert.fail(Assert.java:89)
		at org.junit.Assert.assertTrue(Assert.java:42)
		at org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.lambda$testRemoteCleanupDeleteStale$0(RemoteClusterStateCleanupManagerIT.java:127)
		at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1116)
		... 40 more
	Suppressed: java.lang.AssertionError: Current number of manifest files: 30
		at org.junit.Assert.fail(Assert.java:89)
		at org.junit.Assert.assertTrue(Assert.java:42)
		at org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.lambda$testRemoteCleanupDeleteStale$0(RemoteClusterStateCleanupManagerIT.java:127)
		at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1116)
		... 40 more
	Suppressed: java.lang.AssertionError: Current number of manifest files: 30
		at org.junit.Assert.fail(Assert.java:89)
		at org.junit.Assert.assertTrue(Assert.java:42)
		at org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.lambda$testRemoteCleanupDeleteStale$0(RemoteClusterStateCleanupManagerIT.java:127)
		at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1116)
		... 40 more
```

### Related component

Storage:Remote

### To Reproduce

```
./gradlew ':server:internalClusterTest' --tests "org.opensearch.gateway.remote.RemoteClusterStateCleanupManagerIT.testRemoteCleanupDeleteStale" -Dtests.seed=AF3CFB99BA9F2DEE
```

### Expected behavior

The test must always pass

### Additional Details

**Plugins**
Standard

**Screenshots**

**Host/Environment (please complete the following information):**
 - CI

**Additional context**
- https://build.ci.opensearch.org/job/gradle-check/39719/testReport/junit/org.opensearch.gateway.remote/RemoteClusterStateCleanupManagerIT/testRemoteCleanupDeleteStale/

### LINKED ISSUE #14205: [AUTOCUT] Gradle Check Failure on push to main
Gradle check has failed on push of your commit 990ddc31275e4f8691551aba95d2541b248a87c4.
Please examine the workflow log https://build.ci.opensearch.org/job/gradle-check/40751/.
Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

### PR DESCRIPTION
**Pull request by shiv0408**
### Description
We see that Cleanup interval setting set before the might not be applied before the interval for assert is over.
Added a check to ensure that setting is updated before checking if files are deleted.

Further, addressing some bugs found after merging #14089 
- Addressing a bug in `getManifestCodecVersion` where if path for files have Delimiter `__` we are throwing an error. 
- Changed that expected object in RemoteReadResult to move away from XContent, as this was decided in #13576 but not updated in #14006

Ran the test 1000 times to ensure that flakiness is not there now.

### Related Issues
Resolves #14205
Resolves #13968

### Check List
- [ ] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/gateway/remote/model/RemoteClusterMetadataManifest.java [hunk 1]
```diff
@@ -131,16 +132,18 @@ public ClusterMetadataManifest deserialize(final InputStream inputStream) throws
         return blobStoreFormat.deserialize(blobName, getNamedXContentRegistry(), Streams.readFully(inputStream));
     }
 
-    private int getManifestCodecVersion() {
+    // package private for testing
+    int getManifestCodecVersion() {
         assert blobName != null;
-        String[] splitName = blobName.split(DELIMITER);
+        String[] pathTokens = getBlobPathTokens();
```
Comment (soosinha) [inline server/src/main/java/org/opensearch/gateway/remote/model/RemoteClusterMetadataManifest.java:138]: You can instead use `getBlobFileName()` method to get the file name

Comment (shiv0408) [reply server/src/main/java/org/opensearch/gateway/remote/model/RemoteClusterMetadataManifest.java]: Done

File: server/src/main/java/org/opensearch/gateway/remote/model/RemoteClusterMetadataManifest.java [hunk 2]
```diff
@@ -93,19 +93,27 @@ public String getType() {
...
-            MANIFEST,
-            RemoteStoreUtils.invertLong(clusterMetadataManifest.getClusterTerm()),
-            RemoteStoreUtils.invertLong(clusterMetadataManifest.getStateVersion()),
-            (clusterMetadataManifest.isCommitted() ? COMMITTED : PUBLISHED),
-            RemoteStoreUtils.invertLong(System.currentTimeMillis()),
-            String.valueOf(clusterMetadataManifest.getCodecVersion())
-            // Keep the codec version at last place only, during we read last place to determine codec version.
-        );
-        this.blobFileName = blobFileName;
+        if (blobFileName != null) {
```
Comment (soosinha) [inline server/src/main/java/org/opensearch/gateway/remote/model/RemoteClusterMetadataManifest.java:96]: Dont need to change this logic. You can just use getBlobFileName() where needed

**Review by soosinha — APPROVED** [review]

**Comment by shiv0408** [comment]
> ❌ Gradle check result for https://github.com/opensearch-project/OpenSearch/commit/60f7a88210e4bc9fe07f8695a5d5e7b1fd96d1c7: [FAILURE](https://build.ci.opensearch.org/job/gradle-check/40920/)

[org.opensearch.index.remote.RemoteSegmentTransferTrackerTests.testComputeTimeLagOnUpdate](https://build.ci.opensearch.org/job/gradle-check/40920/testReport/junit/org.opensearch.index.remote/RemoteSegmentTransferTrackerTests/testComputeTimeLagOnUpdate/) - #12639

**Comment by shwetathareja** [comment]
@shiv0408 please update the PR title, it is not just test flakiness

File: server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateAttributesManager.java [hunk 3]
```diff
@@ -121,7 +120,7 @@ public CheckedRunnable<IOException> getAsyncMetadataReadAction(
         LatchedActionListener<RemoteReadResult> listener
     ) {
         final ActionListener actionListener = ActionListener.wrap(
-            response -> listener.onResponse(new RemoteReadResult((ToXContent) response, CLUSTER_STATE_ATTRIBUTE, component)),
+            response -> listener.onResponse(new RemoteReadResult(response, CLUSTER_STATE_ATTRIBUTE, component)),
```
Comment (shwetathareja) [inline server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateAttributesManager.java:123]: do we have test for this?

Comment (shiv0408) [reply server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateAttributesManager.java:123]: We don't have any test for this flow yet. Added one in latest commit

**Comment by andrross** [comment]
> ❌ Gradle check result for [87b05a7](https://github.com/opensearch-project/OpenSearch/commit/87b05a72c53302ed40f8565b79acc411c0864ebf): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/40947/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

@shiv0408 Looks like the "org.opensearch.remotemigration.RemoteStoreMigrationTestCase.testEndToEndRemoteMigration" failure might be a new one. I don't see an open issue for it.

**Comment by prudhvigodithi** [comment]
Hey @andrross and @shiv0408  the automation created an [issue](https://github.com/opensearch-project/OpenSearch/issues/14315) for `RemoteStoreMigrationTestCase` and linked this PR as well, please check.

**Comment by andrross** [comment]
> Hey @andrross and @shiv0408 the automation created an [issue](https://github.com/opensearch-project/OpenSearch/issues/14315) for `RemoteStoreMigrationTestCase` and linked this PR as well, please check.

Thanks @prudhvigodithi. Looks like the `testEndToEndRemoteMigration` failure is not a new one for this PR.

**Comment by prudhvigodithi** [comment]
From this AUTOCUT issue https://github.com/opensearch-project/OpenSearch/issues/14315#issue-2352106645 I can see the `testEndToEndRemoteMigration` has failed on some other PR's and post merge actions.

**Review by shwetathareja — APPROVED** [review]

**Comment by shiv0408** [comment]
> ❌ Gradle check result for https://github.com/opensearch-project/OpenSearch/commit/af9c6be1e1c8fa3c11d74c91d92fad679baabb3f: [FAILURE](https://build.ci.opensearch.org/job/gradle-check/40982/)

[org.opensearch.cache.common.tier.TieredSpilloverCacheStatsIT.testIndicesAndTierLevelAggregation](https://build.ci.opensearch.org/job/gradle-check/40982/testReport/junit/org.opensearch.cache.common.tier/TieredSpilloverCacheStatsIT/testIndicesAndTierLevelAggregation/) - #14281

**Comment by shwetathareja** [comment]
> ❌ Gradle check result for [af9c6be](https://github.com/opensearch-project/OpenSearch/commit/af9c6be1e1c8fa3c11d74c91d92fad679baabb3f): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/40986/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

[org.opensearch.cluster.MinimumClusterManagerNodesIT.testThreeNodesNoClusterManagerBlock](https://build.ci.opensearch.org/job/gradle-check/40986/testReport/junit/org.opensearch.cluster/MinimumClusterManagerNodesIT/testThreeNodesNoClusterManagerBlock_4/)
[org.opensearch.index.IndexServiceTests.testAsyncTranslogTrimTaskOnClosedIndex](https://build.ci.opensearch.org/job/gradle-check/40986/testReport/junit/org.opensearch.index/IndexServiceTests/testAsyncTranslogTrimTaskOnClosedIndex/)

**Comment by shiv0408** [comment]
[org.opensearch.cluster.MinimumClusterManagerNodesIT.testThreeNodesNoClusterManagerBlock](https://build.ci.opensearch.org/job/gradle-check/40986/testReport/junit/org.opensearch.cluster/MinimumClusterManagerNodesIT/testThreeNodesNoClusterManagerBlock_4/) - #10006
[org.opensearch.index.IndexServiceTests.testAsyncTranslogTrimTaskOnClosedIndex](https://build.ci.opensearch.org/job/gradle-check/40986/testReport/junit/org.opensearch.index/IndexServiceTests/testAsyncTranslogTrimTaskOnClosedIndex/) - #11547

**Pull request merged**